addons/web/static/src/views/fields/many2one_auto_save/many2one_auto_save.js

```diff
@@ -0,0 +1,21 @@
+/** @odoo-module **/
+
+import { _lt } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { many2OneField, Many2OneField } from "../many2one/many2one_field";
+
+export class Many2OneAutoSaveField extends Many2OneField {
+    async updateRecord(value) {
+        // Auto-save the record.
+        await this.props.record.save();
+        return await super.updateRecord(...arguments);
+    }
+}
+
+export const many2OneAutoSaveField = {
+    ...many2OneField,
+    component: Many2OneAutoSaveField,
+    displayName: _lt("Many2OneAutoSave"),
```

Comment:
Not sure to understand why we need this :thinking: I added because others widget do that (not a very argument I agree)

Reply:
Each field has a set of descriptive element which allows OWL to render it properly when parsing the XML. I don't think the displayName is necessary however, I think this might be of use for Studio. Doesn't cost much to keep it.